### ISSUE TITLE
Webdav support office - excel

### ISSUE THREAD

**Issue by acosonic** [danmunn/redmine_dmsf#484]
http://www.webdavsystem.com/server/documentation/ms_office_read_only/

Your WebDAV Server must support Class 2. Microsoft Office will lock the item that is being opened for editing.
In the case of Microsoft Office 2007 and earlier your WebDAV server must be available on the site root, it must process OPTIONS and PROPFIND requests.
In case, your server is using Basic authentication your server must be accessed via HTTPS/SSL (at least with self-signed certificate).

**Comment by picman** [comment]
*Your WebDAV Server must support Class 2. Microsoft Office will lock the item that is being opened for editing.*

=> I believe that this is supported in the plugin.

*In the case of Microsoft Office 2007 and earlier your WebDAV server must be available on the site root, it must process OPTIONS and PROPFIND requests.*

=> This condition can't be met as the WebDAV location is always your_domain/dmsf/webdav/....

*In case, your server is using Basic authentication your server must be accessed via HTTPS/SSL (at least with self-signed certificate)*

=> This is true.

What's the problem?

**Comment by picman** [comment]
I've just tested it with MS Office 2010 and it seems that PROPFIND request are send to the root even in this version.
```
I, [2015-12-07T14:34:05.712178 #10048]  INFO -- : Started PROPFIND "/" for 192.168.1.251 at 2015-12-07 14:34:05 +0100
F, [2015-12-07T14:34:05.713731 #10048] FATAL -- : 
ActionController::RoutingError (No route matches [PROPFIND] "/"):
```
So, it works quite fine. A document can be opened, edited and saved via WebDAV directly to DMSF but due to the requests sent to the root you are getting errors nevertheless it works :-(
Another aspect of that is that it works from a mapped drive only. If you click on an Office document in the DMSF web interface, the document is opened as read only because URLs are opened always as read only.

**Comment by acosonic** [comment]
So I tested with office 2013, opening works but only trough SSL mapped drives.

But keeps asking for username/pass when document is accessed with Excel. 

Save does not work, throws this dump in Redmine production log:
```
Started LOCK "/dmsf/webdav/licenciranje/Serveri%20ORACLE-1.xlsx" for 10.20.3.226 at 2015-12-07 16:32:40 +0100

ActiveRecord::RecordNotFound (Couldn't find lock with uuid = <>):
  plugins/redmine_dmsf/app/models/dmsf_lock.rb:66:in `find'
  plugins/redmine_dmsf/lib/redmine_dmsf/webdav/dmsf_resource.rb:429:in `lock'
  /home/redmine/.rvm/gems/ruby-2.2.1/bundler/gems/dav4rack-405a91337016/lib/dav4rack/resource.rb:117:in `method_missing'
  plugins/redmine_dmsf/lib/redmine_dmsf/webdav/resource_proxy.rb:129:in `lock'
  /home/redmine/.rvm/gems/ruby-2.2.1/bundler/gems/dav4rack-405a91337016/lib/dav4rack/resource.rb:117:in `method_missing'
  /home/redmine/.rvm/gems/ruby-2.2.1/bundler/gems/dav4rack-405a91337016/lib/dav4rack/controller.rb:280:in `lock'
  /home/redmine/.rvm/gems/ruby-2.2.1/bundler/gems/dav4rack-405a91337016/lib/dav4rack/handler.rb:30:in `call'
  actionpack (4.2.3) lib/action_dispatch/routing/mapper.rb:51:in `serve'
  actionpack (4.2.3) lib/action_dispatch/journey/router.rb:43:in `block in serve'
  actionpack (4.2.3) lib/action_dispatch/journey/router.rb:30:in `each'
  actionpack (4.2.3) lib/action_dispatch/journey/router.rb:30:in `serve'
  actionpack (4.2.3) lib/action_dispatch/routing/route_set.rb:821:in `call'
  rack-openid (1.4.2) lib/rack/openid.rb:98:in `call'
  request_store (1.0.5) lib/request_store/middleware.rb:9:in `call'
  rack (1.6.4) lib/rack/etag.rb:24:in `call'
  rack (1.6.4) lib/rack/conditionalget.rb:38:in `call'
  rack (1.6.4) lib/rack/head.rb:13:in `call'
  actionpack-xml_parser (1.0.2) lib/action_dispatch/xml_params_parser.rb:16:in `call'
  actionpack (4.2.3) lib/action_dispatch/middleware/params_parser.rb:27:in `call'
  plugins/redmine_dmsf/lib/redmine_dmsf/webdav/no_parse.rb:33:in `call'
  actionpack (4.2.3) lib/action_dispatch/middleware/flash.rb:260:in `call'
  rack (1.6.4) lib/rack/session/abstract/id.rb:225:in `context'
  rack (1.6.4) lib/rack/session/abstract/id.rb:220:in `call'
  actionpack (4.2.3) lib/action_dispatch/middleware/cookies.rb:560:in `call'
  activerecord (4.2.3) lib/active_record/query_cache.rb:36:in `call'
  activerecord (4.2.3) lib/active_record/connection_adapters/abstract/connection_pool.rb:653:in `call'
  actionpack (4.2.3) lib/action_dispatch/middleware/callbacks.rb:29:in `block in call'
  activesupport (4.2.3) lib/active_support/callbacks.rb:84:in `run_callbacks'
  actionpack (4.2.3) lib/action_dispatch/middleware/callbacks.rb:27:in `call'
  actionpack (4.2.3) lib/action_dispatch/middleware/remote_ip.rb:78:in `call'
  actionpack (4.2.3) lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call'
  actionpack (4.2.3) lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
  railties (4.2.3) lib/rails/rack/logger.rb:38:in `call_app'
  railties (4.2.3) lib/rails/rack/logger.rb:20:in `block in call'
  activesupport (4.2.3) lib/active_support/tagged_logging.rb:68:in `block in tagged'
  activesupport (4.2.3) lib/active_support/tagged_logging.rb:26:in `tagged'
  activesupport (4.2.3) lib/active_support/tagged_logging.rb:68:in `tagged'
  railties (4.2.3) lib/rails/rack/logger.rb:20:in `call'
  actionpack (4.2.3) lib/action_dispatch/middleware/request_id.rb:21:in `call'
  rack (1.6.4) lib/rack/methodoverride.rb:22:in `call'
  rack (1.6.4) lib/rack/runtime.rb:18:in `call'
  activesupport (4.2.3) lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
  actionpack (4.2.3) lib/action_dispatch/middleware/static.rb:116:in `call'
  rack (1.6.4) lib/rack/sendfile.rb:113:in `call'
  railties (4.2.3) lib/rails/engine.rb:518:in `call'
  railties (4.2.3) lib/rails/application.rb:165:in `call'
  railties (4.2.3) lib/rails/railtie.rb:194:in `public_send'
  railties (4.2.3) lib/rails/railtie.rb:194:in `method_missing'
  /usr/lib/ruby/vendor_ruby/phusion_passenger/rack/thread_handler_extension.rb:97:in `process_request'
  /usr/lib/ruby/vendor_ruby/phusion_passenger/request_handler/thread_handler.rb:160:in `accept_and_process_next_request'
  /usr/lib/ruby/vendor_ruby/phusion_passenger/request_handler/thread_handler.rb:113:in `main_loop'
  /usr/lib/ruby/vendor_ruby/phusion_passenger/request_handler.rb:416:in `block (3 levels) in start_threads'
  /usr/lib/ruby/vendor_ruby/phusion_passenger/utils.rb:113:in `block in create_thread_and_abort_on_exception'
```

**Comment by acosonic** [comment]
But it works like charm (excel) trough this:
http://totalcmd.net/plugring/webdav.html

**Comment by picman** [comment]
Solved in scope of #587.
